### DISCUSSION TITLE
How to configure middleware with https

### DISCUSSION THREAD

**Discussion by pradeepdev-1995** [fastapi/fastapi#8456] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.middleware("https")
```

### Description

I would like to configure https with middleware like
```@app.middleware("https")```
But it shows an error like

**_AssertionError: Currently only middleware("http") is supported._**

So how to configure https with middleware in fastAPI?

### Operating System

Linux

### FastAPI Version

fastapi==0.75.0

### Python Version

3.6.9

**Comment by jgould22** [comment, marked as answer]
Here are FastAPIs docs on https https://fastapi.tiangolo.com/deployment/https/

The current recommended approach is to keep the https complexity outside of fast api using a proxy. Some options are listed on that page

 - Traefik (that can also handle certificate renewals)
 - Caddy (that can also handle certificate renewals)
 - Nginx
 - HAProxy
